python/whylogs/api/logger/segment_processing.py

```diff
@@ -129,15 +131,19 @@ def _log_segment(
     pandas: Optional[pd.DataFrame] = None,
     row: Optional[Mapping[str, Any]] = None,
     segment_cache: Optional[SegmentCache] = None,
+    segment_key_values: Optional[Dict[str, str]] = None,
 ) -> Dict[Segment, Any]:
     segments: Dict[Segment, Any] = {}
     pandas, row = _pandas_or_dict(obj, pandas, row)
     if partition.filter:
+        # TODO: segment_key_values don't apply to filter since they're explicitely added
         pandas, row = _filter_inputs(partition.filter, pandas, row)
     if partition.simple:
         columns = partition.mapper.col_names if partition.mapper else None
         if columns:
-            _process_simple_partition(partition.id, schema, segments, columns, pandas, row, segment_cache)
+            _process_simple_partition(
```

Comment:
Should we allow empty columns if `segment_key_values` are specified?

Reply:
Its an edge case, not sure I understand how we get an empty column set with a simple segmentation partition, but if we do have an explicit segment via `segment_key_values` it would be good to handle this like the explicit segmentation route.

Reply:
I knocked up some code to handle the case, but I'm inclined to think of this as a user error -- use an unsegmented schema and `segment_key_values` for this case. Maybe throw or log an error here?

Reply:
I'm ok with logging an error here, hinting on the correct usage pattern.